src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ReflectionXmlFormatWriter.cs

```diff
@@ -210,13 +211,25 @@ internal void ReflectionWriteCollection(XmlWriterDelegator xmlWriter, object obj
                     IEnumerator nonGenericDictEnumerator = (IEnumerator)new CollectionDataContract.DictionaryEnumerator(((IDictionary)obj).GetEnumerator());
                     enumerator = nonGenericDictEnumerator;
                 }
+                else if (collectionDataContract.Kind == CollectionKind.GenericCollection)
+                {
+                    elementType = collectionDataContract.ItemType;
+                }
                 else
```

Comment:
Yes, this is as expected. The new 'else if' block deals with one collection kind specially. The 'else' would deal with other kinds.